examples/tuners/dngo_tuner/dngo_tuner.py

```diff
@@ -0,0 +1,116 @@
+import random
+import torch
+import numpy as np
+
+from nni.tuner import Tuner
+import nni.parameter_expressions as parameter_expressions
+from torch.distributions import Normal
+from pybnn import DNGO
+
+def random_archi_generator(nas_ss, random_state):
+    '''random
+    '''
+    chosen_arch = {}
+    for key, val in nas_ss.items():
+        assert val['_type'] in ['choice', 'uniform'], \
+            "Random NAS Tuner only receives NAS search space whose _type is 'choice' or 'uniform'"
+        if val['_type'] == 'choice':
+            choices = val['_value']
+            index = random_state.randint(len(choices))
+            # check values type
+            if type(choices[0]) == int or type(choices[0]) == float:
+                chosen_arch[key] = choices[index]
+            else:
+                chosen_arch[key] = index
+        elif val['_type'] == 'uniform':
+            chosen_arch[key] = random.uniform(val['_value'][0], val['_value'][1])
+        elif val['_type'] == 'randint':
+            chosen_arch[key] = random_state.randint(
+                    val['_value'][0], val['_value'][1], size=1)
+        elif val['_type'] == 'quniform':
+            chosen_arch[key] = parameter_expressions.quniform(
+                val['_value'][0], val['_value'][1], val['_value'][2], random_state)
+        elif val['_type'] == 'loguniform':
+            chosen_arch[key] = parameter_expressions.loguniform(
+                val['_value'][0], val['_value'][1], random_state)
+        elif val['_type'] == 'qloguniform':
+            chosen_arch[key] = parameter_expressions.qloguniform(
+                val['_value'][0], val['_value'][1], val['_value'][2], random_state)
+
+        else:
+            raise ValueError('Unknown key %s and value %s' % (key, val))
+    return chosen_arch
+
+class DngoTuner(Tuner):
```

Comment:
@QuanluZhang Is DNGO an example of customized tuner or a built-in tuner?

Reply:
DNGO is meant to be a built-in tuner. Let me check how to make it a built-in tuner.